### DISCUSSION TITLE
Error when calling Dask groupby with custom aggregate

### DISCUSSION THREAD

**Discussion by bihshteinCiti** [dask/dask#7028] [Q&A]
The following error occurs when running the below example test, what am I doing wrong (using dask 2.3.0)?

Error: Exception('Column(s) id already selected',)

Code:

```python
import unittest 
import dask 
import pandas as pd 
import dask.dataframe as dd

class TestDaskCustomAgg(unittest.TestCase):

def mode(self, x):
    val = pd.Series.mode(x)
    if val.empty:
        return np.NaN
    return val[0]

def test_get_transactions(self):
    df = dask.datasets.timeseries()
    custom_agg = dd.Aggregation('custom_agg', agg=lambda x: self.mode(x), chunk=lambda x0: self.mode(x0))
    df.groupby('name').agg(custom_agg).compute()
```

**Comment by jsignell** [comment]
I would recommend that you look closely at the examples and try to reason about that the inputs and outputs of each function should be. You might find these docs: https://docs.dask.org/en/latest/dataframe-groupby.html#aggregate useful if you haven't read them yet.

**Reply by bihshteinCiti** [reply]
Yes I have read those but couldn't understand the difference between agg and chunk parameters as originally (with pandas) I only passed single lambda expression

**Reply by jsignell** [reply]
Right so the chunk function operates on each partition of the dataframe and then the agg function operates on the results of the chunk operations.
